lib/otel/segments/consumer.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+module.exports = createConsumerSegment
+
+// Notes:
+// + https://github.com/open-telemetry/semantic-conventions/blob/v1.24.0/docs/messaging/messaging-spans.md
+// + We probably want to inspect `messaging.system` so that we can generate
+// attributes according to our own internal specs.
+
+const Transaction = require('../../transaction/')
+const { DESTINATIONS, TYPES } = Transaction
+
+const {
+  // SEMATTRS_MESSAGING_SYSTEM,
+  SEMATTRS_MESSAGING_DESTINATION,
+  SEMATTRS_MESSAGING_OPERATION
+} = require('@opentelemetry/semantic-conventions')
+
+function createConsumerSegment({ agent, otelSpan }) {
+  const transaction = new Transaction(agent)
+  transaction.type = TYPES.BG
+
+  const operation = otelSpan.attributes[SEMATTRS_MESSAGING_OPERATION]
+  const destination = otelSpan.attributes[SEMATTRS_MESSAGING_DESTINATION] ?? 'unknown'
+  const segmentName = `OtherTransaction/consumer/${operation}/${destination}`
+
+  transaction.trace.attributes.addAttribute(
```

Comment:
there's more logic in the [subscribe-consume](https://github.com/newrelic/node-newrelic/blob/main/lib/shim/message-shim/subscribe-consume.js#L174) for adding other segment and trace attributes

Reply:
A good test, specific to amqplib but should apply to all but obviously the destination kind and destination vary: https://github.com/newrelic/node-newrelic/blob/main/test/versioned/amqplib/amqp-utils.js#L103

Reply:
I don't understand this one. Is there a document somewhere that clearly lists out the attribute names and expected values that are required?